### PR TITLE
8337410: The makefiles should set problemlist and adjust timeout basing on the given VM flags

### PR DESCRIPTION
**Pull request by lmesnik**
There jtreg tests have several additional problemlists 
ProblemList-Xcomp.txt
ProblemList-generational-zgc.txt
ProblemList-zgc.txt
Each of them is bound to corresponding execution mode (Xcomp/ZGC) and it makes sense to treat them like standard problemlist when tests are executed with -Xcomp or ZGC enabled.
Currently, it is needed to set them manually and it is often forgotten. So engineers waste time analyzing known failures.

Additionally, the **default** timeoutFactor is increased when Xcomp is enabled because of slowness of this mode. 

The jtreg allows to add the same problemlist twice so it is not needed to update any execution system that set problemlists.

Later it might makes sens to set 'JTREG_ALL_OPTIONS' by asking java about actually set mode. So it is possible to adjust  options for fastdebug/slowdebug/product modes and/or different options that are set during compilation (saying different default GC).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8337410](https://bugs.openjdk.org/browse/JDK-8337410): The makefiles should set problemlist and adjust timeout basing on the given VM flags (**Enhancement** - P4)

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20430/head:pull/20430` \
`$ git checkout pull/20430`

Update a local copy of the PR: \
`$ git checkout pull/20430` \
`$ git pull https://git.openjdk.org/jdk.git pull/20430/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20430`

View PR using the GUI difftool: \
`$ git pr show -t 20430`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20430.diff">https://git.openjdk.org/jdk/pull/20430.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20430#issuecomment-2263708838)

### CODE REVIEW

File: make/RunTests.gmk [hunk 1]
```diff
@@ -837,6 +835,24 @@ define SetupRunJtregTestBody
     $1_JTREG_BASIC_OPTIONS += $$(addprefix $$(JTREG_PROBLEM_LIST_PREFIX), $$($1_JTREG_PROBLEM_LIST))
   endif
 
+  JTREG_ALL_OPTIONS := $$(JTREG_JAVA_OPTIONS) $$(JTREG_VM_OPTIONS)
+
+  JTREG_AUTO_PROBLEM_LISTS :=
+  JTREG_AUTO_TIMEOUT_FACTOR := 4
+
+  ifneq ($$(findstring -Xcomp, $$(JTREG_ALL_OPTIONS)), )
+    JTREG_AUTO_PROBLEM_LISTS := $$(JTREG_AUTO_PROBLEM_LISTS) ProblemList-Xcomp.txt
```
Comment (magicus) [inline make/RunTests.gmk:844]: Please use `+=` instead.

File: make/RunTests.gmk [hunk 2]
```diff
@@ -837,6 +835,24 @@ define SetupRunJtregTestBody
...
+    JTREG_AUTO_TIMEOUT_FACTOR ?= 10
```
Comment (magicus) [inline make/RunTests.gmk:845]: Since you assign this to be 4 a couple of lines up, this code will never do anything.

Comment (magicus) [reply make/RunTests.gmk:845]: Did you test that this patch works as intended?

File: make/RunTests.gmk [hunk 3]
```diff
@@ -837,6 +835,24 @@ define SetupRunJtregTestBody
...
+  endif
+
+  ifneq ($$(findstring -XX:+UseZGC, $$(JTREG_ALL_OPTIONS)), )
+    ifneq ($$(findstring -XX:-ZGenerational, $$(JTREG_ALL_OPTIONS)), )
```
Comment (magicus) [inline make/RunTests.gmk:849]: Is this the only way that zgc can be run in a non-generational mode? I understand the idea to automatically pick the right set of problem list files, I just want to make sure it is robust.

File: make/RunTests.gmk (hunk 2 above)
Comment (lmesnik) [reply make/RunTests.gmk]: Thanks for catching.
Yes, the problem is that our system always set timeout factor. So now I also run manual testing using make to verify that correct factor is used.

File: make/RunTests.gmk (hunk 3 above)
Comment (lmesnik) [reply make/RunTests.gmk:849]: Yes, I checked that  -XX:+UseZGC -XX:-ZGenerational and didn't find any other way to enable this mode.

Comment (magicus) [reply make/RunTests.gmk:849]: Great, thanks!

**Review by magicus — APPROVED** [review]
This looks good now. Thanks!

**Comment by dholmes-ora** [comment]
Doesn't the use of multiple problem-list files like this greatly increase the risk of running into duplicate PL entries which are not handled correctly by jtreg?

**Comment by dholmes-ora** [comment]
Never mind. The problem is only with multiple entries in the same file apparently: [CODETOOLS-7903657](https://bugs.openjdk.org/browse/CODETOOLS-7903657)

**Pull request closed without merging**